### PR TITLE
fixed hard delete and add getsoftdeleteddashboard

### PR DESCRIPTION
**Pull request by lucychen-grafana**
FOR HACKATHON PROJECT, REVIEWS WILL BE MADE WHEN THE FEATURE REACHES MAIN.

### CODE REVIEW

File: pkg/api/dashboard.go [hunk 1]
```diff
@@ -272,33 +272,39 @@ func (hs *HTTPServer) getDashboardHelper(ctx context.Context, orgID int64, id in
 	return queryResult, nil
 }
 
+func (hs *HTTPServer) GetSoftDeletedDashboard(ctx context.Context, orgID int64, uid string) (*dashboards.Dashboard, response.Response) {
```
Comment (evictorero) [inline pkg/api/dashboard.go:275]: Is this method in the API necessary? You can call the `GetSoftDeletedDashboard` from the service directly or if you want to add this method here it should be private.

File: pkg/services/dashboards/database/database.go [hunk 2]
```diff
@@ -640,6 +640,26 @@ func (d *dashboardStore) GetDashboardsByPluginID(ctx context.Context, query *das
 	}
 	return dashboards, nil
 }
+func (d *dashboardStore) GetSoftDeletedDashboard(ctx context.Context, query *dashboards.GetDashboardQuery) (*dashboards.Dashboard, error) {
+	var queryResult *dashboards.Dashboard
+	err := d.store.WithDbSession(ctx, func(sess *db.Session) error {
+
+		dashboard := dashboards.Dashboard{OrgID: query.OrgID, UID: query.UID}
+		mustCols := []string{}
```
Comment (evictorero) [inline pkg/services/dashboards/database/database.go:648]: What is the purpose here of mustCols?

Comment (lucychen-grafana) [reply pkg/services/dashboards/database/database.go:648]: It's used to append the query search below. I referenced the original [GetDashboard](https://github.com/grafana/grafana/blob/main/pkg/services/dashboards/database/database.go#L812) method. I could write it out all in one line as well

Comment (evictorero) [reply pkg/services/dashboards/database/database.go:648]: It shouldn't be necessary unless you specify some columns, in this case is empty so you can remove it before the `.Get`

File: pkg/api/dashboard.go (hunk 1 above)
Comment (evictorero) [reply pkg/api/dashboard.go:275]: In that case you can change the first letter to lowercase since it is private and will be used only in this package.

Comment (lucychen-grafana) [reply pkg/api/dashboard.go]: I decided to simplify the code and remove that function

**Review by evictorero — APPROVED** [review]

**Pull request merged**